src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -311,10 +310,9 @@ bool ObjectSynchronizer::quick_notify(oopDesc* obj, Thread* self, bool all) {
 // Note that we can't safely call AsyncPrintJavaStack() from within
 // quick_enter() as our thread state remains _in_Java.
 
-bool ObjectSynchronizer::quick_enter(oop obj, Thread* self,
+bool ObjectSynchronizer::quick_enter(oop obj, JavaThread* current,
                                      BasicLock * lock) {
-  assert(!SafepointSynchronize::is_at_safepoint(), "invariant");
```

Comment:
Did you mean to delete this assert()? If so, why?
Update: because the next assert rules out being at a safepoint because you
can't have a thread in _thread_in_Java at a safepoint. Okay...